packages/vm/src/evm/precompiles/01-ecrecover.ts

```diff
@@ -1,10 +1,11 @@
 import { setLengthLeft, setLengthRight, ecrecover, publicToAddress, BN } from 'ethereumjs-util'
 import { PrecompileInput } from './types'
 import { OOGResult, ExecResult } from '../evm'
-const assert = require('assert')
 
 export default function (opts: PrecompileInput): ExecResult {
-  assert(opts.data)
+  if (!opts.data) {
+    throw new Error('opts.data is undefined')
```

Comment:
I've written a long thread on our current error management which also touches the current problem field of changing error msgs, see here: https://github.com/ethereumjs/ethereumjs-monorepo/pull/1469#issuecomment-1023016428 (where we likely want to improve on this round of breaking releases).

But since these error messages haven't been before anyhow: can we directly take something a bit more high-level and expressive here? Something like `No data provided for the ECRECOVER precompile` (feel free to improve).

Reply:
(same for the other precompiles of course)